js/modules/k6/crypto/crypto.go

```diff
@@ -39,23 +39,70 @@ import (
 	"github.com/dop251/goja"
 
 	"go.k6.io/k6/js/common"
+	"go.k6.io/k6/js/modules"
 )
 
-type Crypto struct{}
+type (
+	// RootModule is the global module instance that will create module
+	// instances for each VU.
+	RootModule struct{}
 
-type Hasher struct {
-	ctx context.Context
+	// Crypto represents an instance of the crypto module.
+	Crypto struct {
+		vu  modules.VU
+		obj *goja.Object
+	}
+)
+
+var (
+	_ modules.Module   = &RootModule{}
+	_ modules.Instance = &Crypto{}
+)
+
+// New returns a pointer to a new RootModule instance.
+func New() *RootModule {
+	return &RootModule{}
+}
+
+// NewModuleInstance implements the modules.Module interface to return
+// a new instance for each VU.
+func (*RootModule) NewModuleInstance(vu modules.VU) modules.Instance {
+	rt := vu.Runtime()
+	o := rt.NewObject()
+	mi := &Crypto{vu: vu, obj: o}
+
+	mustExport := func(name string, value interface{}) {
+		if err := mi.obj.Set(name, value); err != nil {
+			common.Throw(rt, err)
+		}
+	}
 
-	hash hash.Hash
+	mustExport("createHash", mi.CreateHash)
+	mustExport("createHMAC", mi.CreateHMAC)
+	mustExport("hmac", mi.HMAC)
+	mustExport("md4", mi.MD4)
+	mustExport("md5", mi.MD5)
+	mustExport("randomBytes", mi.RandomBytes)
+	mustExport("ripemd160", mi.Ripemd160)
+	mustExport("sha1", mi.SHA1)
+	mustExport("sha256", mi.SHA256)
+	mustExport("sha384", mi.SHA384)
+	mustExport("sha512", mi.SHA512)
+	mustExport("sha512_224", mi.SHA512224)
+	mustExport("sha512_256", mi.SHA512256)
+	mustExport("hexEncode", mi.HexEncode)
```

Comment:
I kind of feel like this will be better of just as `map[string]interface{}` in the `Exports`. Is there anything, in particular, we are winning by doing it this way?